integration/Substra/plot_substrafl_flamby.py

```diff
@@ -0,0 +1,427 @@
+        z.write(filepath, arcname=filepath.name)
+
+metric_key = clients[ALGO_ORG_ID].add_algo(objective)
+
+# %%
+# Registering dataset
+# ===================
+#
+# A :ref:`documentation/concepts:Dataset` is composed of an **opener**, which is a Python script with the instruction
+# of *how to load the data* from the files in memory, and a **description markdown** file.
+
+
+train_datasets_dict = dict()
+for org_id in ORGS_ID:
+    train_datasets_dict[org_id] = DatasetSpec(
+        name="FLamby",
+        type="torchDataset",
+        data_opener=assets_directory / "dataset" / f"opener_train_{org_id}.py",
+        description=assets_directory / "dataset" / "description.md",
+        permissions=permissions,
+        logs_permission=permissions,
+    )
+
+# %%
+# Train and test data nodes
+# =========================
+#
+# The :ref:`documentation/concepts:Dataset` object itself does not contain the data. The proper asset to access them
+# is the **datasample asset**.
+#
+# A **datasample** contains a local path to the data, and the key identifying the :ref:`documentation/concepts:Dataset`
+# it is based on, in order to have access to the proper `opener.py` file.
+#
+# To interface with Flamby, the utilization of the opener and datasample is a bit bypassed here.
+# Substra is a library built to be deployed on a real federated setup, where the path to the data is provided to
+# the opener in order to load and read them in a personalized way directly on the organization.
+# As we directly load a torch dataset from Flamby, the folders parameters is unused and the path usually leading to
+# the data will point out a empty folder in this example.
+#
+# Now that all our :ref:`documentation/concepts:Assets` are well defined, we can create
+# :ref:`substrafl_doc/api/nodes:TrainDataNode` and :ref:`substrafl_doc/api/nodes:TestDataNode` to gathered the
+# :ref:`documentation/concepts:Dataset` and the **datasamples** on the specified organization.
+
+train_data_nodes = list()
+test_data_nodes = list()
+
+for ind, org_id in enumerate(ORGS_ID):
+    client = clients[org_id]
+
+    # Add the dataset to the client to provide access to the opener in each organization.
+    train_dataset_key = client.add_dataset(train_datasets_dict[org_id])
+    assert train_dataset_key, "Missing data manager key"
+
+    # Add the training data on each organization.
+    data_sample = DataSampleSpec(
+        data_manager_keys=[train_dataset_key],
+        test_only=False,
+        path=pathlib.Path.cwd() / "empty",
+    )
+    train_datasample_key = client.add_data_sample(
+        data_sample,
+        local=True,
+    )
+
+    # Create the Train Data Node (or training task) and save it in a list
+    train_data_node = TrainDataNode(
+        organization_id=org_id,
+        data_manager_key=train_dataset_key,
+        data_sample_keys=[train_datasample_key],
+    )
+    train_data_nodes.append(train_data_node)
+
+# Add the testing data.
+
+client = clients[ALGO_ORG_ID]
```

Comment:
There should be 6 centers for testing not 1

Reply:
For this one, for now in the example we are testing on the pooled test dataset. Do you mean that you would prefer testing the model on each test subset ?

Reply:
@jeandut up

Reply:
Yes the proper way to test would be on each test set for each center.

Reply:
The example have been updated, ready for a second review @jeandut :)